util/SqliteMigrations/HelperScripts/2024-03-22_00_EnableOrgsCollectionEnhancements.sql

```diff
@@ -0,0 +1,151 @@
+    WHERE "O"."FlexibleCollections" = 0 AND "G"."AccessAll" = 1;
+
+-- Step 2: AccessAll migration for OrganizationUsers
+    -- Create a temporary table to store the OrganizationUsers with AccessAll = 1
+    CREATE TEMPORARY TABLE IF NOT EXISTS "TempUsersAccessAll" AS
+    SELECT "OU"."Id" AS "OrganizationUserId",
+           "OU"."OrganizationId"
+    FROM "OrganizationUser" "OU"
+    INNER JOIN "Organization" "O" ON "OU"."OrganizationId" = "O"."Id"
+    WHERE "O"."FlexibleCollections" = 0 AND "OU"."AccessAll" = 1;
+
+-- Step 3: For all OrganizationUsers with Manager role or 'EditAssignedCollections' permission update their existing CollectionUsers rows and insert new rows with [Manage] = 1
+-- and finally update all OrganizationUsers with Manager role to User role
+    -- Create a temporary table to store the OrganizationUsers with Manager role or 'EditAssignedCollections' permission
+    CREATE TEMPORARY TABLE IF NOT EXISTS "TempUserManagers" AS
+    SELECT "OU"."Id" AS "OrganizationUserId",
+           CASE WHEN "OU"."Type" = 3 THEN 1 ELSE 0 END AS "IsManager"
+    FROM "OrganizationUser" "OU"
+    INNER JOIN "Organization" "O" ON "OU"."OrganizationId" = "O"."Id"
+    WHERE "O"."FlexibleCollections" = 0 AND
+        ("OU"."Type" = 3 OR
+         ("OU"."Type" = 4 AND
+          "OU"."Permissions" IS NOT NULL AND
+          JSON_VALID("OU"."Permissions") AND JSON_EXTRACT(ou."Permissions", '$.editAssignedCollections') = 1));
+
+-- Step 1
+    -- Update existing rows in "CollectionGroups"
+    UPDATE "CollectionGroups"
+    SET
+        "ReadOnly" = 0,
+        "HidePasswords" = 0,
+        "Manage" = 0
+    WHERE EXISTS (
+        SELECT 1
+        FROM "Collection" "C"
+        INNER JOIN "TempGroupsAccessAll" "TG" ON "CollectionGroups"."GroupId" = "TG"."GroupId"
+        WHERE "CollectionGroups"."CollectionId" = "C"."Id" AND C."OrganizationId" = "TG"."OrganizationId"
+    );
+
+    -- Insert new rows into "CollectionGroups"
+    INSERT INTO "CollectionGroups" ("CollectionId", "GroupId", "ReadOnly", "HidePasswords", "Manage")
+    SELECT "C"."Id", "TG"."GroupId", 0, 0, 0
+    FROM "Collection" "C"
+    INNER JOIN "TempGroupsAccessAll" "TG" ON "C"."OrganizationId" = "TG"."OrganizationId"
+    LEFT JOIN "CollectionGroups" "CG" ON "CG"."CollectionId" = "C"."Id" AND "CG"."GroupId" = "TG"."GroupId"
+    WHERE "CG"."CollectionId" IS NULL;
+
+    -- Update "Group" to clear "AccessAll" flag and update "RevisionDate"
+    UPDATE "Group"
+    SET "AccessAll" = 0, "RevisionDate" = CURRENT_TIMESTAMP
+    WHERE "Id" IN (SELECT "GroupId" FROM "TempGroupsAccessAll");
+
+-- Step 2
+    -- Update existing rows in "CollectionUsers"
+    UPDATE "CollectionUsers"
+    SET
+        "ReadOnly" = 0,
+        "HidePasswords" = 0,
+        "Manage" = 0
+    WHERE "CollectionId" IN (
+        SELECT "C"."Id"
+        FROM "Collection" "C"
+        INNER JOIN "TempUsersAccessAll" "TU" ON "C"."OrganizationId" = "TU"."OrganizationId"
```

Comment:
Again, this is only joining on Organization.Id when I think it should also be joining with the temp table on CollectionUser.OrganizationUserId.

Please double check for this potential issue anywhere else.

Reply:
I also updated the 3 scripts here.